cvm-securekey-release-app/AttestationUtil.cpp

```diff
@@ -566,10 +632,21 @@ bool Util::doSKR(const std::string &attestation_url, const std::string &nonce, s
         std::string attest_token(Util::GetMAAToken(attestation_url, nonce));
         TRACE_OUT("MAA Token: %s", attest_token.c_str());
 
-        std::string akvMsiToken = std::move(Util::GetIMDSToken(client_id));
-        TRACE_OUT("AkvMsiToken: %s", akvMsiToken.c_str());
-        json json_object = json::parse(akvMsiToken.c_str());
-        std::string access_token = json_object["access_token"].get<std::string>();
+        // Get Akv access token either using IMDS or Service Principal
+        std::string access_token;
+        switch (akv_credential_source)
+        {
+        case Util::AkvCredentialSource::EnvServicePrincipal:
+            access_token = std::move(Util::GetAADToken());
+            break;
+        case Util::AkvCredentialSource::Imds:
+        default:
+            access_token = std::move(Util::GetIMDSToken());
+            break;
+        }

```

Comment:
The switch statement is not required in that case since Util::GetAADToken() does not take credential source as parameter.